prompt_toolkit/shortcuts/progress_bar/base.py

```diff
@@ -361,18 +364,49 @@ def item_completed(self) -> None:
 
     @property
     def done(self) -> bool:
+        """Whether a counter has been completed.
+
+        Done counter have been stopped (see stop) and removed depending on
+        remove_when_done value.
+
+        Contrast this with stop. A stopped counter may be terminated before
+        100% completion. A done counter has reached its 100% completion.
+        """
         return self._done
 
     @done.setter
     def done(self, value: bool) -> None:
         self._done = value
-
-        # If done then store the stop_time, otherwise clear.
-        self.stop_time = datetime.datetime.now() if value else None
+        self.stop = value
 
         if value and self.remove_when_done:
             self.progress_bar.counters.remove(self)
 
+    @property
+    def stop(self) -> bool:
+        """Whether a counter has been stopped.
+
+        Stopped counters no longer have increasing time_elapsed. This distinction is
+        also used to prevent the Bar formatter with unknown totals from continuing to run.
+
+        A stopped counter (but not done) can be used to signal that a given counter has
+        encountered an error but allows other counters to continue
+        (e.g. download X of Y failed). Given how only done counters are removed
+        (see remove_when_done) this can help aggregate failures from a large number of
+        successes.
+
+        Contrast this with done. A done counter has reached its 100% completion.
+        A stopped counter may be terminated before 100% completion.
+        """
+        return self._stop
```

Comment:
I think we can remove the ``self._stop`` variable and do ``return self.stop_time is not None`` here.

Reply:
Agreed